src/main/java/codechicken/nei/api/ShortcutInputHandler.java

```diff
@@ -247,6 +252,25 @@ private static boolean openOverlayRecipe(ItemStack stackover) {
         return false;
     }
 
+    private static boolean saveFavoriteTree(ItemStack stackover) {
+        final Point mouse = GuiDraw.getMousePosition();
+
+        if (ItemPanels.itemPanel.contains(mouse.x, mouse.y)
+                || ItemPanels.itemPanel.historyPanel.contains(mouse.x, mouse.y)) {
```

Comment:
I see this check is repeated in several places. Does ItemPanel already provide something like it?
If not, it may be worth extracting a helper method to avoid duplication, for example:

```java
private static boolean itemPanelContains(ItemPanel itemPanel, float x, float y) {
    return itemPanel.contains(x, y) || itemPanel.historyPanel.contains(x, y);
}
```

Reply:
this was changed in the PR i am working on now